rules/custom.yml

```diff
@@ -0,0 +1,29 @@
+# https://unicode-org.github.io/icu/userguide/boundaryanalysis/break-rules.html
+
+# Keep hyphens
+keep_hyphens: |
+    $Hyphen = [\p{Pd}];
+    ($ALetterPlus | $Hebrew_Letter | $Numeric)  $ExFm* $Hyphen $ExFm* ($ALetterPlus | $Hebrew_Letter | $Numeric);
+
+# Keep URLs
+keep_url: |
+    ^(([h][t][t][p][s]?\:\/\/([w][w][w]\.)?)|([w][w][w]\.))[-a-zA-Z0-9@:%_\+\.~#=]+(\/[-a-zA-Z0-9@\:%_\+\.~#?\&=]+)*[\/]?;
```

Comment:
@odelmarcelle  I think this should be literal 
```
"('http' | 'https') '://' [-a-zA-Z0-9@:%_+.~#=]+;"
```

Reply:
You're right, I wasn't aware we could specify sequences of literals like this. I've rewritten and enhanced the rule.